tests/test_operations.py

```diff
@@ -409,20 +443,363 @@ def paginate(self):
+            # Monkeypatch the lambda client
+            with mock.patch("dss.operations.params.lambda_client") as lam:
+                # The lambda_list func in params.py
+                # calls get_deployed_lambas, which calls
+                # lam.get_function() using daemon folder names
+                # (this function is called only to ensure no exception
+                # is thrown)
+                lam.get_function = mock.MagicMock(return_value=None)
+                # Next we call get_deployed_lambda_environment(),
+                # which calls lam.get_function_configuration
+                # (this returns the mocked new env vars json)
+                lam.get_function_configuration = mock.MagicMock(return_value=lam_new_env)
+
+                # Used to specify a lambda by name
+                stage = os.environ["DSS_DEPLOYMENT_STAGE"]
+
+                # Non-JSON fmt, no lambda name specified
+                with CaptureStdout() as output:
+                    ssm_params.lambda_list(
+                        [],
+                        argparse.Namespace(
+                            lambda_name=None,
+                            json=False
+                        )
+                    )
+                self.assertIn(f"{testvar_name}={testvar_value}", output)
+
+                # Non-JON fmt, lambda name specified
+                with CaptureStdout() as output:
+                    ssm_params.lambda_list(
+                        [],
+                        argparse.Namespace(
+                            lambda_name=f"dss-{stage}",
+                            json=False
+                        )
+                    )
+                self.assertIn(f"{testvar_name}={testvar_value}", output)
+
+                # JSON fmt, no lambda name specified
+                with CaptureStdout() as output:
+                    ssm_params.lambda_list(
+                        [],
+                        argparse.Namespace(
+                            lambda_name=None,
+                            json=True
+                        )
+                    )
+                all_lam_envs = json.loads("\n".join(output))
+                for _, lam_env in all_lam_envs.items():
+                    self.assertIn(testvar_name, lam_env)
+
+                # JSON fmt, lambda name specified
+                with CaptureStdout() as output:
+                    ssm_params.lambda_list(
+                        [],
+                        argparse.Namespace(
+                            lambda_name=f"dss-{stage}",
+                            json=True
+                        )
+                    )
+                all_lam_envs = json.loads("\n".join(output))
+                keys = list(all_lam_envs.keys())
+                self.assertTrue(len(keys) == 1)
+                self.assertIn(testvar_name, all_lam_envs[keys[0]])
+
+        with self.subTest("Update existing lambda parameters"):
```

Comment:
Is there a reason that these context managers are nested instead of being nested a la

```
with self.subTest("Update existing lambda parameters"), \
    mock.patch("dss.operations.params.ssm_client") as ssm, \
    mock.patch("dss.operations.params.lambda_client") as lam:
```
?